cmd/git/main.go

```diff
@@ -493,6 +493,13 @@ func checkCredentials() (credentialType, error) {
 	case hasUsername && hasPassword && isHTTPSURL:
 		return typeUsernamePassword, nil
 
+	case hasUsername && hasPassword && !isHTTPSURL:
+		return typeUndef, &ExitError{
+			Code:    110,
+			Message: shpgit.AuthUnexpectedHTTP.ToMessage(),
+			Reason:  shpgit.AuthUnexpectedHTTP,
+		}
```

Comment:
You're here checking if the URL is not HTTPS and then return an error stating that HTTP is unexpected. What happens if the protocol is neither http nor https but maybe `htps` ?

Reply:
Yeah, good point. I tried to be too minimalistic here I guess.